src/main/java/StringAddCalculator.java

```diff
@@ -0,0 +1,39 @@
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class StringAddCalculator {
+
+    private StringAddCalculator() {}
+
+    public static int splitAndSum(String text) {
+        if (checkNullOrEmpty(text)) return 0;
+        String[] numbers = splitTextToNumbers(text);
+        return sum(numbers);
+    }
+
+    private static boolean checkNullOrEmpty(String text) {
+        return text == null || text.equals("");
+    }
+
+    private static String[] splitTextToNumbers(String text) {
+        Matcher m = Pattern.compile("//(.)\n(.*)").matcher(text);
+        if (m.find()) {
+            String customDelimiter = m.group(1);
+            return m.group(2).split(customDelimiter);
+        }
+        return text.split(",|:");
+    }
+
+    private static int sum(String[] numbers) {
+        int result = 0;
+        for (String number : numbers) {
+            result = result + checkNegative(Integer.parseInt(number));
+        }
+        return result;
+    }
```

Comment:
```suggestion
    private static int sum(String[] numbers) {
        int result = 0;
        for (String number : numbers) {
            int num = Integer.parseInt(number);
            checkNegative(num);
            result += num;
        }
        return result;
    }
```
checkNegative는 음수인지 체크하고 예외만 던지는 로직만 담고있어도 될 것 같아요
변환은 Integer.parseInt가 해주니까요